src/client/components/button/button.component.js

```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+import './button.style.css';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+
+export default function Button({ children, type, size, onClick, toggled }) {
+  const btnClass = classNames({
+    btn: true,
+    'btn-primary': type === 'primary',
+    'btn-secondary': type === 'secondary',
+    'btn-danger': type === 'danger',
+    'btn-toggle': type === 'toggle',
+    'btn-large': size,
+    toggled, // true if the toggle button is pressed and held down
+  });
```

Comment:
I prefer to add "static" classes that are not generated based on logic to be declared outside the object like this:

```suggestion
  const btnClass = classNames('btn', {
    'btn-primary': type === 'primary',
    'btn-secondary': type === 'secondary',
    'btn-danger': type === 'danger',
    'btn-toggle': type === 'toggle',
    'btn-large': size,
    toggled, // true if the toggle button is pressed and held down
  });
```